### DISCUSSION TITLE
How to Write Vector<JSON>Directly into the Output Stream and Maintain Format

### DISCUSSION THREAD

**Discussion by Cukipa** [nlohmann/json#4034] [Q&A]
`vector<json> js;
ofstream ofs(output.json);`

I want to use the following very simple sentences to achieve output effects

`ofs<<js;`

But the output is concentrated on one line, and I hope to achieve the effect of 'json::dump(3)'

Is there any way to maintain simplicity and formatting?

**Comment by nlohmann** [comment]
You can use `std::setw(3)` to set the desired indentation, see https://json.nlohmann.me/api/operator_gtgt/

**Comment by antilneeraj** [comment]
You can follow these steps:

1. Obtain a reference to the output stream where you want to write the JSON data. This could be a file stream, network stream, or any other suitable output stream implementation.

2. Convert the `Vector<JSON>` into a formatted string representation. Depending on the programming language and libraries you are using, this step may vary. Most programming languages provide libraries for working with JSON, which usually include methods for converting JSON objects or arrays to string representations.

3. Write the formatted string representation of the `Vector<JSON>` into the output stream. Use the appropriate method provided by the output stream implementation to write the data. This method typically takes a string as input.

Below is an example in Java that demonstrates these steps using the Jackson library for JSON manipulation:

```java
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.ObjectWriter;

import java.io.OutputStream;
import java.util.Vector;

public class JsonWriterExample {
    public static void writeVectorToJsonStream(Vector<JSON> jsonData, OutputStream outputStream) throws JsonProcessingException {
        // Create an instance of the Jackson ObjectMapper
        ObjectMapper objectMapper = new ObjectMapper();

        // Create an ObjectWriter with appropriate formatting options
        ObjectWriter objectWriter = objectMapper.writerWithDefaultPrettyPrinter();

        // Convert the Vector<JSON> to a formatted JSON string
        String jsonString = objectWriter.writeValueAsString(jsonData);

        // Write the JSON string to the output stream
        try {
            outputStream.write(jsonString.getBytes());
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
```
